modules/drivers/bigquery-cloud-sdk/src/metabase/driver/bigquery_cloud_sdk.clj

```diff
@@ -0,0 +1,312 @@
+(ns metabase.driver.bigquery-cloud-sdk
+  (:require [clojure.set :as set]
+            [clojure.string :as str]
+            [clojure.tools.logging :as log]
+            [medley.core :as m]
+            [metabase.driver :as driver]
+            [metabase.driver.bigquery-cloud-sdk.common :as bigquery.common]
+            [metabase.driver.bigquery-cloud-sdk.params :as bigquery.params]
+            [metabase.driver.bigquery-cloud-sdk.query-processor :as bigquery.qp]
+            [metabase.query-processor.error-type :as error-type]
+            [metabase.query-processor.store :as qp.store]
+            [metabase.query-processor.timezone :as qp.timezone]
+            [metabase.query-processor.util :as qputil]
+            [metabase.util :as u]
+            [metabase.util.i18n :refer [tru]]
+            [metabase.util.schema :as su]
+            [schema.core :as s])
+  (:import com.google.auth.oauth2.ServiceAccountCredentials
+           [com.google.cloud.bigquery BigQuery BigQuery$JobOption BigQuery$TableListOption BigQuery$TableOption
+                                      BigQueryOptions DatasetId EmptyTableResult Field FieldValue FieldValueList
+                                      QueryJobConfiguration Schema Table TableId TableResult]
+           java.io.ByteArrayInputStream
+           java.util.Collections))
+
+(driver/register! :bigquery-cloud-sdk, :parent :sql)
+
+
+;;; +----------------------------------------------------------------------------------------------------------------+
+;;; |                                                     Client                                                     |
+;;; +----------------------------------------------------------------------------------------------------------------+
+
+(def ^:private bigquery-scope
+  "The scope to use for executing BigQuery requests; see:
+  `https://cloud.google.com/bigquery/docs/samples/bigquery-auth-drive-scope`.
+  Unclear if this can be sourced from the `com.google.cloud.bigquery` package directly."
+  "https://www.googleapis.com/auth/bigquery")
+
+(defn- ^ServiceAccountCredentials database->service-account-credential
+  "Returns a `ServiceAccountCredentials` (not scoped) for the given DB, from its service account JSON."
+  ^ServiceAccountCredentials [{{:keys [^String service-account-json]} :details
```

Comment:
nit seems you have the typehint twice.

Reply:
Oops, removed the one before the symbol name, as per: https://clojure.org/reference/java_interop#typehints